### DISCUSSION TITLE
Does Curl --krb work with HTTP / HTTPS instead of --negotiate

### DISCUSSION THREAD

**Discussion by ganeshkamath89** [curl/curl#10103] [Q&A] [answered]
Hi Curl team,
We have a requirement to support plain Kerberos implementation with KRB5 instead of using Kerberos Negotiate. As our server runs in space constrained environment.

negotiate tries Kerberos, and falls back to NTLM.

From Curl docmentation it looks like **--krb** option supports FTP. Does it also support HTTP/HTTPS?

![image](https://user-images.githubusercontent.com/2648522/208019107-405e71ee-8d53-4973-8b7d-dd8122855254.png)

When we try curl --krb with HTTP it is not working, so we wanted to check with the community whether the support exists or if there is some reason for not supporting this with HTTP/HTTPS.

Command: 
**Working negotiate command:**
`curl -v --negotiate private -P <port> -u <domain-name>\<user-name>:<password> http://<domain-system>:<port>/<HTML-Page>.htm`

**Non-working kerberos command:**
`curl -v --krb private -P <port> -u <domain-name>\<user-name>:<password> http://<domain-system>:<port>/<HTML-Page>.htm`

We use Curl 7.86.0.
We have tried and verified observation from Wrindows and Linux.

Is there an example FTP Server which supports Kerberos authentication?

**Comment by bagder** [comment, marked as answer]
The man page lists **FTP** pretty clearly as the protocol for `--krb`, so no it does not.

**Reply by ganeshkamath89** [reply]
Hi @bagder, can you please tell me more on why that decision is made.

Why is direct Kerberos not used with other protocols?

**Reply by bagder** [reply]
It's not a "decision", it is a document that describes how things currently work. It works this way because this is what has been implemented. Feel most welcome to enhance the functionality.
